pykafka/producer.py

```diff
@@ -180,56 +331,128 @@ def _get_partition_msgs(partition_id, req):
             attempt += 1
             if attempt < self._max_retries:
                 time.sleep(self._retry_backoff_ms / 1000)
-                self._produce(to_retry, attempt)
+                # we have to call _produce here since the broker/partition
+                # target for each message may have changed
+                for _ in self._produce(to_retry, attempt):
+                    pass
```

Comment:
Yeah, this is not great. Could use some wrapper function to make this code cleaner, and to automatically iterate the generator for users who want to ignore the yielded values.